code/game/objects/items/storage/fancy.dm

```diff
@@ -17,24 +17,34 @@
 	icon_state = "donutbox6"
 	name = "donut box"
 	var/icon_type = "donut"
+	var/spawn_type
+	var/spawn_number
 
-	update_icon()
-		icon_state = "[icon_type]box[contents.len]"
+/obj/item/storage/fancy/Initialize(mapload, ...)
+	if(spawn_type)
+		can_hold = list(spawn_type) // must be set before parent init for typecacheof
+	. = ..()
+	if(spawn_type)
+		for(var/i in 1 to spawn_number)
+			new spawn_type(src)
 
-	remove_from_storage(obj/item/W, atom/new_location)
-		. = ..()
-		if(.)
-			update_icon()
+/obj/item/storage/fancy/update_icon()
+	icon_state = "[icon_type]box[contents.len]"
 
+/obj/item/storage/fancy/remove_from_storage(obj/item/W, atom/new_location)
+	. = ..()
+	if(.)
+		update_icon()
 
-	examine(mob/user)
-		..()
-		if(contents.len <= 0)
-			to_chat(user, "There are no [src.icon_type]s left in the box.")
-		else if(contents.len == 1)
-			to_chat(user, "There is one [src.icon_type] left in the box.")
-		else
-			to_chat(user, "There are [src.contents.len] [src.icon_type]s in the box.")
+
+/obj/item/storage/fancy/examine(mob/user)
+	. = ..()
+	if(contents.len <= 0)
+		to_chat(user, "There are no [src.icon_type]s left in the box.")
+	else if(contents.len == 1)
```

Comment:
This could easily be turned into one with the ? operator. Also unneccessary src's